mongomock/aggregate.py

```diff
@@ -509,6 +509,89 @@ def _handle_lookup_stage(in_collection, database, options):
     return in_collection
 
 
+def _handle_graph_lookup_stage(in_collection, database, options):
+    if not isinstance(options.get('maxDepth', 0), six.integer_types):
+        raise OperationFailure(
+            "Argument 'maxDepth' to $graphLookup must be a number")
+    if not isinstance(options.get('restrictSearchWithMatch', {}), dict):
+        raise OperationFailure(
+            "Argument 'restrictSearchWithMatch' to $graphLookup must be a Dictionary")
+    if not isinstance(options.get('depthField', ''), six.string_types):
+        raise OperationFailure(
+            "Argument 'depthField' to $graphlookup must be a string")
+    for operator in ('from', 'startWith', 'connectFromField', 'connectToField', 'as'):
+        if operator not in options:
+            raise OperationFailure(
+                "Must specify '%s' field for a $graphLookup" % operator)
+        if not isinstance(options[operator], six.string_types):
+            raise OperationFailure(
```

Comment:
Actually startWith could be an array, according to the doc. Please add a NotImplementedError for that case.

Reply:
You're right - good catch.

Reply:
Added support for any expression in startWith.